homeassistant/components/habitica/button.py

```diff
@@ -0,0 +1,148 @@
+        ),
+    ),
+    HabiticaButtonEntityDescription(
+        key=HabitipyButtonEntity.REVIVE,
+        translation_key=HabitipyButtonEntity.REVIVE,
+        press_fn=(lambda coordinator: coordinator.api["user"]["revive"].post()),
+        available_fn=(lambda data: data.user["stats"]["hp"] == 0),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: HabiticaConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up buttons from a config entry."""
+
+    coordinator = entry.runtime_data
+
+    async_add_entities(
+        HabiticaButton(coordinator, description) for description in BUTTON_DESCRIPTIONS
+    )
+
+
+class HabiticaButton(CoordinatorEntity[HabiticaDataUpdateCoordinator], ButtonEntity):
+    """Representation of a Habitica button."""
+
+    _attr_has_entity_name = True
+    entity_description: HabiticaButtonEntityDescription
+
+    def __init__(
+        self,
+        coordinator: HabiticaDataUpdateCoordinator,
+        entity_description: HabiticaButtonEntityDescription,
+    ) -> None:
+        """Initialize a Habitica button."""
+        super().__init__(coordinator)
+        if TYPE_CHECKING:
+            assert coordinator.config_entry.unique_id
+        self.entity_description = entity_description
+        self._attr_unique_id = (
+            f"{coordinator.config_entry.unique_id}_{entity_description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer=MANUFACTURER,
+            model=NAME,
+            name=coordinator.config_entry.data[CONF_NAME],
+            configuration_url=coordinator.config_entry.data[CONF_URL],
+            identifiers={(DOMAIN, coordinator.config_entry.unique_id)},
+        )
+
+    async def async_press(self) -> None:
+        """Handle the button press."""
+        try:
+            await self.entity_description.press_fn(self.coordinator)
+        except ClientResponseError as e:
+            if e.status == HTTPStatus.TOO_MANY_REQUESTS:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="setup_rate_limit_exception",
+                ) from e
+            if e.status == HTTPStatus.UNAUTHORIZED:
+                raise ServiceValidationError(
+                    translation_domain=DOMAIN,
+                    translation_key="service_call_unallowed",
+                ) from e
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="service_call_exception",
+            ) from e
+        else:
+            await self.coordinator.async_refresh()
+
+    @property
+    def available(self) -> bool:
+        """Is entity available."""
+        if self.entity_description.available_fn:
+            return self.entity_description.available_fn(self.coordinator.data)
+        return True
```

Comment:
We need to check the state with the coordinator first:
```suggestion
        if not super().available:
        	return False
        if self.entity_description.available_fn:
            return self.entity_description.available_fn(self.coordinator.data)
        return True
```

Reply:
Ah, thanks, didn't know that 👍